templates/two-columns/two-columns.js

```diff
@@ -0,0 +1,60 @@
+import { loadCSS } from '../../scripts/lib-franklin.js';
+
+/**
+ * Checks if the ratio is different from 50-50 and applies the ratio to its column respectively
+ * @param {HTMLElement} columnOne - first column section
+ * @param {HTMLElement} columnTwo - second column section
+ */
+function setRatio(columnOne, columnTwo) {
+  const ratio = document.querySelector('meta[name="column-ratio"]').content;
+  if (ratio) {
+    const isDefault = ratio.includes('50');
+    if (!isDefault) {
+      const [firstRatio, secondRatio] = ratio.split('-');
+      const root = document.querySelector(':root');
+
+      root.style.setProperty('--first-ratio', `${firstRatio}%`);
+      root.style.setProperty('--second-ratio', `${secondRatio}%`);
+
+      columnOne.classList.add('first-column');
+      columnTwo.classList.add('second-column');
+    }
+  }
+}
+
+/**
+ * Modify the page so that it has a two-column template for the first two sections
+ */
+export default function buildAutoBlocks() {
+  const main = document.querySelector('main');
+  const firstSectionIsHero = [...main.children][0].children[0].classList.contains('hero');
+
+  // remove empty div sections
+  [...main.children].forEach((node) => {
+    const hasChildren = [...node.children].length > 0;
+    if (!hasChildren) {
+      node.remove();
+    }
+  });
+
+  const columnOne = firstSectionIsHero ? [...main.children][1] : [...main.children][0];
+  const columnTwo = firstSectionIsHero ? [...main.children][2] : [...main.children][1];
+  const newSection = document.createElement('section');
+
+  const observer = new MutationObserver((list) => {
```

Comment:
do we really need the observer here? what exactly are we waiting for?

Reply:
is needed to wait until the build cycle ends to move the sections. if not, the builder doesn't find them and then doesn't work

Reply:
which other script is moving the sections?

Reply:
franklin-lib and scripts are who do the build cycle. this template script has to wait until the build cycle ends. otherwise, the 2 sections don't build properly

Reply:
the Franklin build cycle is in scripts/lib-franklin.js and scripts/scripts.js. the template is executed before this build cycle, so it has to wait until the cycle ends. the decorateMain function has the main element calls to build the page.
[https://github.com/Netcentric/cn-website/blob/c706b982a36f716ecc031eebc78affe08627b136/scripts/scripts.js#L266](url)
in this function, _buildAutoBlocks_ must be before _decoratesSection_, even if it is an async call, because if not the sections don't build properly. for this reason, the two-columns .js has to wait until the next function call ends

Reply:
I still don't like the waiting, please create a separate ticket so we can have a look into getting rid of it.